keps/sig-node/4009-add-cdi-devices-to-device-plugin-api/README.md

```diff
@@ -0,0 +1,853 @@
+components? What are the guarantees? Make sure this is in the test plan.
+
+Consider the following in developing a version skew strategy for this
+enhancement:
+- Does this enhancement involve coordinating behavior in the control plane and
+  in the kubelet? How does an n-2 kubelet without this feature available behave
+  when this feature is used?
+- Will any other components on the node change? For example, changes to CSI,
+  CRI or CNI may require updating that component before the kubelet.
+-->
+
+## Production Readiness Review Questionnaire
+
+<!--
+
+Production readiness reviews are intended to ensure that features merging into
+Kubernetes are observable, scalable and supportable; can be safely operated in
+production environments, and can be disabled or rolled back in the event they
+cause increased failures in production. See more in the PRR KEP at
+https://git.k8s.io/enhancements/keps/sig-architecture/1194-prod-readiness.
+
+The production readiness review questionnaire must be completed and approved
+for the KEP to move to `implementable` status and be included in the release.
+
+In some cases, the questions below should also have answers in `kep.yaml`. This
+is to enable automation to verify the presence of the review, and to reduce review
+burden and latency.
+
+The KEP must have a approver from the
+[`prod-readiness-approvers`](http://git.k8s.io/enhancements/OWNERS_ALIASES)
+team. Please reach out on the
+[#prod-readiness](https://kubernetes.slack.com/archives/CPNHUMN74) channel if
+you need any help or guidance.
+-->
+
+### Feature Enablement and Rollback
+
+<!--
+This section must be completed when targeting alpha to a release.
+-->
+
+###### How can this feature be enabled / disabled in a live cluster?
+
+<!--
+Pick one of these and delete the rest.
+
+Documentation is available on [feature gate lifecycle] and expectations, as
+well as the [existing list] of feature gates.
+
+[feature gate lifecycle]: https://git.k8s.io/community/contributors/devel/sig-architecture/feature-gates.md
+[existing list]: https://kubernetes.io/docs/reference/command-line-tools-reference/feature-gates/
+-->
+
+This change is not behind a feature gate. It is adding an optional
+field to a stable API. This mirrors what was done in [KEP-1884](https://github.com/kubernetes/enhancements/pull/1884) when
+adding a `cpu_ids` field to the existing PodResources API.
+
+<!-- Furthermore, the only code that would be featured-gated in the Device Manager would be the code to propagate this field from the `AllocateResponse` to the
+corresponding field in the CRI. -->
+
+###### Does enabling the feature change any default behavior?
+
+No. Device Plugins need to be updated to make use of the new field.
+<!--
+Any change of default behavior may be surprising to users or break existing
+automations, so be extremely careful here.
+-->
+
+###### Can the feature be disabled once it has been enabled (i.e. can we roll back the enablement)?
+
+No.
```

Comment:
Also does not make sense for an alpha feature.

Reply:
The entire change is adding a field in the device-plugin protobuf and then passing that untouched through the kubelet to the CRI. It is hard for me to wrap my head around what it would mean to "disable" this.

Reply:
> It is hard for me to wrap my head around what it would mean to "disable" this.

Disable means "it is impossible to use this feature through Kubernetes without setting the feature gate."  Thus when the feature gate is disabled, the field would be explicitly unset (or cause an error on use).